src/app/ReadClient.cpp

```diff
@@ -491,6 +497,7 @@ CHIP_ERROR ReadClient::OnMessageReceived(Messaging::ExchangeContext * apExchange
         ChipLogProgress(DataManagement, "SubscribeResponse is received");
         VerifyOrExit(apExchangeContext == mExchange.Get(), err = CHIP_ERROR_INCORRECT_STATE);
         err = ProcessSubscribeResponse(std::move(aPayload));
+        MATTER_LOG_METRIC_END(Tracing::kMetricDeviceSubscriptionSetup, err);
```

Comment:
If something inside ProcessSubscribeResponse() before it calls `MoveToState` fails, then we will do the MATTER_LOG_METRIC_END here _and_ in the resulting Close() call, right?

I wonder whether it would be better to put ` MATTER_LOG_METRIC_END(Tracing::kMetricDeviceSubscriptionSetup, CHIP_NO_ERROR);` in either MoveToState when moving to the `SubscriptionActive` state or at the one place where we move to that state.  That way we unambiguously do the "ended with error" in Close() and "ended with success" on the codepath that does that.

Reply:
The backend needs to handle multiple end or begin calls. The usual way is to capture the first of these events.